src/controllers/shortener.controller.ts

```diff
@@ -90,3 +121,61 @@ export const deleteUrl = async (req: Request<{ id: string }>, res: Response) =>
     return res.status(INTERNAL_SERVER_ERROR).json({ error: 'Internal Server Error' });
   }
 };
+
+interface UpdateURLRequest extends Request {
+  body: {
+    objectid: string;
+    originalUrl?: string;
+    customName?: string;
+  };
+}
+
+export const updateURL = async (req: UpdateURLRequest, res: Response) => {
+  const { objectid, originalUrl, customName } = req.body;
```

Comment:
The id should be a route paramater

```suggestion
  const { originalUrl, customName } = req.body;
```